### PR TITLE
Added SIGINT catch for exit controller. Added log job on progress eve…

### PR DESCRIPTION
**Pull request by erlstanleylee**
…nt. Update visibility of of closeAllDaemons and Finish to Client interface.

### CODE REVIEW

File: bin/controller.ts [hunk 1]
```diff
@@ -28,9 +28,20 @@ const job1: Job = new NormalJob('first', new Set([job4, job2]))
...
+})
 once(client, 'done').then(clientDone).catch(console.error)
 
+process.on('SIGINT', () => {
+	console.log('\nClosing Daemons\n')
+
+	client.closeAllDaemonsAndFinish()
+
+	console.log('Exit Process\n')
+	process.exit()
```
Comment (rvighne) [inline bin/controller.ts:42]: It shouldn't be necessary to force-exit the process. Ideally, once you close all the sockets, you just wait for the event loop will run out. What happens if you remove this line?

File: src/Client/Client.ts [hunk 2]
```diff
@@ -51,4 +51,9 @@ export interface Client extends TypedEmitter<ClientEvents> {
 	 * @param port - port number of daemon on the host
 	 */
 	introduce(host: string, port: number): void
+
+	/**
+	 * Close all daemons
+	 */
+	closeAllDaemonsAndFinish(): void
```
Comment (rvighne) [inline src/Client/Client.ts:58]: Could you rename this method to be shorter in both classes? E.g. `quit`.

**Review by rvighne — CHANGES_REQUESTED** [review]
Good work.

File: src/Client/Client.ts (hunk 2 above)
Comment (erlstanleylee) [reply src/Client/Client.ts:58]: Will do.

File: bin/controller.ts (hunk 1 above)
Comment (erlstanleylee) [reply bin/controller.ts:42]: Yup do not need to force-exit. Removed this line.

**Review by rvighne — APPROVED** [review]

File: bin/controller.ts [hunk 3]
```diff
@@ -28,9 +28,19 @@ const job1: Job = new NormalJob('first', new Set([job4, job2]))
 const job5: Job = new NormalJob('fifth', new Set([job1]))
 
 const client: Client = new Http2Client(new HeapJobOrderer([job5]))
-client.on('progress', console.log)
+client.on('progress', (FinishedJob) => {
```
Comment (Rohanator9000) [inline bin/controller.ts:31]: Why is the variable `FinishedJob` uppercase?

Comment (erlstanleylee) [reply bin/controller.ts:31]: My mistake. Will change.

**Review by yashlala — APPROVED** [review]

**Pull request merged**